x/pool-incentives/abci.go

```diff
@@ -0,0 +1,32 @@
+package pool_incentives
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/c-osmosis/osmosis/x/pool-incentives/keeper"
+)
+
+func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper) {
+	if ctx.BlockHeight() > 1 {
+		// WARNING: The order of how modules interact with the default distribution module matters if the distribution module is used in a similar way to:
```

Comment:
Technically, this will also grab the tx fees that were paid in OSMO right?

Would it help to have it noted down somewhere in the distribution module how much was minted?

Or maybe we can even have a hook in the mint module with the newly minted amount, that calls a function in this module, so it only has to run this every time tokens are minted, instead of every block?

This way, basically, this module doesn't touch distribution (distribution is really specifically `staking-distribution` and instead interfaces only with `mint` directly)

Reply:
Looking at it again, it seems like you can know the minted coin amount from the mint module. So the pool incentives module should be able to distribute the incentive amount which excludes the fees.

But even then, this wouldn't solve the issue where the modules BeginBlock order needs to be followed, and therefore the values have to be set relatively.

Reply:
Here's PR for converting beginblocker to hooks. Let's switch chat into this PR.
https://github.com/osmosis-labs/osmosis/pull/167